climada/hazard/tc_tracks_synth.py

```diff
@@ -452,6 +452,14 @@ def _calc_land_decay(hist_tracks, land_geom, s_rel=True, check_plot=False,
     Returns:
         v_rel (dict(category: A)), p_rel (dict(category: (S, B)))
     """
+    
+    if len(hist_tracks) < 10:
+        LOGGER.warning('Using only %s historical tracks to calculate '
+                       'landfall decay. Consider using a larger number of '
+                       'tracks from which you sub-select the tracks that '
+                       'are relevant to your application for a more robust '
+                       'calculation of landfall decay.', len(hist_tracks))
+
```

Comment:
Users might not realize that unlike for e.g. missing values estimations in read_ibtracs_netcdf, the landfall parameters are derived solely based on the historical tracks in the TCTracks object. Hence if less than 10 tracks are there, is warns the user. Is 10 a good threshold? Or what should be the criterion to warn the user?

Reply:
Well, the question should be investigated right? ... what is the error made from having 10 tracks? Does it depend on the geographical area?

Reply:
Adding this warning is actually not related to the issues this PR fixes, but I added it because I think it is useful to warn users here. Here is my reasoning: There are several cases in TCTracks where a fit is applied to the data. For instance, in `TCTracks.read_ibtracs_netcdf` with `estimate_missing=True`, a fit is applied to fill in missing values. However, in that case the fit has been calculated beforehand using the full IBTrACS data from 1980-2020 and the resulting parameter values are provided in a static way, e.g. in `estimate_pressure`:
https://github.com/CLIMADA-project/climada_python/blob/038cfc40ebb83c6f114557ed7576bbba022e9860/climada/hazard/tc_tracks.py#L1464
This ensures that the same "model" is used whichever input are passed to the function.

In the case of landfall decay for synthetic tracks, there is no pre-computed fitted parameters. Instead, the parameters are fitted from all the historical tracks that are passed to `calc_perturbed_trajectories`. Therefore, the parameters used to characterize landfall decay depend on the content of your TCTracks object when calling the function.

Personally, here is the approach I would propose:
1. Calculating landfall decay parameters using all tracks from 1980-2020 (for consistency with read_ibtracs_netcdf). However I would do that by basin.
2. Hard-code these values as a dictionary, and use these to apply landfall decay.

That being said, this is a separate topic that is not part of this PR, and investing time in this is not worthwhile in case the synthetic tracks model would be replaced by something more physically-based. Perhaps we could create a separate issue on this. In the meantime I propose this warning, although I'd suggest to strongly increase the threshold (there is one set of parameter per Saffir-Simpson category, which can only be calculated for tracks that make landfall at that category, so perhaps we could increase the threshold to 100 or something like this.

Reply:
OK, I see. I think your solution is valid and is also desired for the current implementation. 

Note that there will most probably never be a physics-based random sampling module. Doing such models is very expensive both in scientific knowledge, work time and computation time. This is also not our expertise nor is it the goal of CLIMADA. There are several research groups and commercial agents working on precisely this. What should be reflected upon is whether a better statistical model can be made. However, also this is actually a whole research subject on its own. The best is to have data APIs. It is not the aim, nor are there the capacities to make a full-fledged tropical cyclone statistical model.

Hence, doing the calibration is the best way. I agree that the per basin is also a good idea.

Concerning the threshold, why 100? Do you have a basis for this number (or an educated guess)? Why not 1000?